File: example/exam03Service/app/src/main/java/com/gbox3d/exam03service/MainActivity.java
package com.gbox3d.exam03service;

import androidx.appcompat.app.AppCompatActivity;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;



public class MainActivity extends AppCompatActivity {

    public final static String TAG  = "exam03";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d(TAG,"start activity");

        ((Button)findViewById(R.id.button_start)).setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        Intent intent = new Intent(MainActivity.this,MyService.class);
                        intent.putExtra("message", "This is my first message!");

                        startService(intent);


                    }
                }
        );

        ((Button)findViewById(R.id.button_stop)).setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        stopService(new Intent(MainActivity.this,MyService.class));

                    }
                }
        );

        ((Button)findViewById(R.id.button_send_serv)).setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        Intent intent = new Intent("custom-event-name");
                        intent.putExtra("message", "This is my first message!");

                        Log.d(TAG,"broad cast");

                        LocalBroadcastManager.getInstance(MainActivity.this).sendBroadcast(intent);
                    }
                }
        );
    }
}

File: ex02_hello/app/src/main/java/com/gbox3d/ex02_hello/MainActivity.kt
package com.gbox3d.ex02_hello

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.gbox3d.ex02_hello.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var mBinder: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)

        mBinder = ActivityMainBinding.inflate(layoutInflater)
        val view = mBinder.root

        setContentView(view)

        mBinder.btnOk.setOnClickListener{
            val name = mBinder.edName.text

            Log.d("Main", "${name}")

            mBinder.tvName.text = name
        }


    }
}